### ISSUE TITLE
multiplication became broken in noncommutative rings recently

### ISSUE THREAD

**Issue by mikestillman** [Macaulay2/M2#2811]
I am surprised this became broken (or rather, I'm surprised that there are no tests to make sure it is working correctly).
```
needsPackage "AssociativeAlgebras"
A = QQ <|x,y|>
-- these are fine
x*y
y*x
M = matrix {{x}}
N = matrix {{y}}
-- these are not
y * M
M * y
```

The problem appears to have been added in commit 36fdb90298, and no test revealed that it broke non-commutative multiplication.

I think we need to change the code in `m2/matrix.m2`: 
```
Matrix * RingElement := (m,r) -> r*m
```

back to what it was:

```
Matrix * RingElement := (m,r) -> (
     r = promote(r,ring m);
     map(target m, source m, reduce(target m, raw m * raw r)))
```

**Comment by pzinn** [comment]
oops! my bad

**Comment by moorewf** [comment]
I think the fix is slightly different -- I think one should take the code from the multiplication in the other order (RingElement * Matrix) and reverse the order of multiplication.  this is because the RingElement * Matrix code contains the fix that @pzinn was fixing with 36fdb90.

**Comment by mahrud** [comment]
Probably worth checking everything in `methods symbol*`.

Incidentally, I noticed that `RingElement * MonomialIdeal` is terribly wrong!! Try `(x + y) * monomialIdeal x` and it gives `x^2`!
